wgengine/packet/packet.go

```diff
@@ -130,39 +175,113 @@ func ipChecksum(b []byte) uint16 {
 	return uint16(^ac)
 }
 
-var put16 = binary.BigEndian.PutUint16
-var put32 = binary.BigEndian.PutUint32
+// IPHeader represents a header of an IP packet.
+type IPHeader struct {
+	SrcIP IP
+	DstIP IP
+	IPID  uint16
+}
+
+func writeIPHeader(header IPHeader, proto byte, out []byte) {
```

Comment:
1. Why is proto a parameter here instead of a field of IPHeader?
2. How about making this function a method on IPHeader?
3. `write` is an odd term to use here in Go-land. It suggests the destination is an io.Writer. Indeed if you made it a method on IPHeader with this name, it would look like it is writing *to* the IPHeader object. I think this would be better called Fill, or you could change it to an Append-style method that takes a []byte and returns a []byte.